test/integration/crud/insert.test.ts

```diff
@@ -0,0 +1,1640 @@
+    //       const db = client.db(configuration.db);
+    //       const collection = db.collection('bson_types_insert');
+    //
+    //       const document = {
+    //         symbol: new BSONSymbol('abcdefghijkl'),
+    //         objid: new ObjectId('abcdefghijkl'),
+    //         double: new Double(1),
+    //         binary: new Binary(Buffer.from('hello world')),
+    //         minkey: new MinKey(),
+    //         maxkey: new MaxKey(),
+    //         code: new Code('function () {}', { a: 55 })
+    //       };
+    //
+    //       collection.insert(document, configuration.writeConcernMax(), function (err, result) {
+    //         expect(err).to.not.exist;
+    //         test.ok(result);
+    //
+    //         collection.findOne({ symbol: new BSONSymbol('abcdefghijkl') }, function (err, doc) {
+    //           expect(err).to.not.exist;
+    //           test.equal('abcdefghijkl', doc.symbol.toString());
+    //
+    //           collection.findOne({ objid: new ObjectId('abcdefghijkl') }, function (err, doc) {
+    //             expect(err).to.not.exist;
+    //             test.equal('6162636465666768696a6b6c', doc.objid.toString());
+    //
+    //             collection.findOne({ double: new Double(1) }, function (err, doc) {
+    //               expect(err).to.not.exist;
+    //               test.equal(1, doc.double);
+    //
+    //               collection.findOne(
+    //                 { binary: new Binary(Buffer.from('hello world')) },
+    //                 function (err, doc) {
+    //                   expect(err).to.not.exist;
+    //                   test.equal('hello world', doc.binary.toString());
+    //
+    //                   collection.findOne({ minkey: new MinKey() }, function (err, doc) {
+    //                     expect(err).to.not.exist;
+    //                     test.ok(doc.minkey._bsontype === 'MinKey');
+    //
+    //                     collection.findOne({ maxkey: new MaxKey() }, function (err, doc) {
+    //                       expect(err).to.not.exist;
+    //                       test.ok(doc.maxkey._bsontype === 'MaxKey');
+    //
+    //                       collection.findOne(
+    //                         { code: new Code('function () {}', { a: 55 }) },
+    //                         function (err, doc) {
+    //                           expect(err).to.not.exist;
+    //                           test.ok(doc != null);
+    //                           client.close(done);
+    //                         }
+    //                       );
+    //                     });
+    //                   });
+    //                 }
+    //               );
+    //             });
+    //           });
+    //         });
+    //       });
+    //     });
+    //   }
+    // });
+
+    it('handles BSON type inserts', async function () {
+      const configuration = this.configuration;
+
+      const document = {
+        symbol: new BSONSymbol('abcdefghijkl'),
```

Comment:
I added this type from another test `handleBSONTypeInsertsCorrectly`, as those 2 tests had overlap - having just one with full set of types would be enough.